aggregator/src/binary_utils/job_driver.rs

```diff
@@ -117,13 +114,17 @@ where
 
         // Set up state for the job driver run.
         let sem = Arc::new(Semaphore::new(self.max_concurrent_job_workers));
-        let mut job_discovery_delay = Duration::ZERO;
+
+        let mut next_run_instant = Instant::now();
+        if !self.job_discovery_interval.is_zero() {
+            next_run_instant += thread_rng().gen_range(Duration::ZERO..self.job_discovery_interval);
```

Comment:
I think we should apply jitter in a tighter range. If I configure job discovery interval to be 60 seconds, I'd be surprised to learn that an interval of 1 ms is just as likely as 60 s. Maybe choose a random value in the range `[0.9 * discovery_interval, 1.1 * discovery_interval)`?

Reply:
This isn't jitter, though: this is effectively selecting a uniformly random time in the the job discovery window for this task to attempt to acquire jobs. This will (probably) distribute job-acquiration attempts from different instances of this deployment uniformly across the job discovery interval.

Though maybe jitter would be the better solution? (IMO, I'm OK with this approach.)

Reply:
OK, I got confused because the PR description says "random jitter strategy used by the job creator". However I notice that the implementation here is what `aggregation_job_creator` already does. That's been working fine (AFAIK) so I'm comfortable using that strategy in the job drivers, too. I guess let's just make sure not to call it jitter.

Reply:
@branlwyd is correct--sorry I didn't update the PR description when I realized the aggregator job creator's strategy is not to jitter but to do offset.